src/toucan2/tools/transformed.clj

```diff
@@ -281,20 +281,15 @@
 
 ;;;; before insert
 
-;;; TODO -- this shares a lot of code with [[transform-update-changes]]
 (defn- transform-insert-rows [[first-row :as rows] k->transform]
   {:pre [(map? first-row) (map? k->transform)]}
-  ;; all rows should have the same keys, so we just need to look at the keys in the first row
-  (let [row-xforms (for [k     (keys first-row)
-                         :let  [xform (get k->transform k)]
-                         :when xform]
-                     (fn [row]
-                       (update row k (fn [v]
-                                       (if (some? v)
-                                         (xform v)
-                                         v)))))
-        row-xform  (apply comp row-xforms)]
-    (map row-xform rows)))
+  (let [x-forms (for [[k transform] k->transform]
```

Comment:
should we do the same for `transform-update-changes`? instead of loop through the row and checking if `k->transform` exists a transformation function, we'll build out an `xform` from `k->transform`.

a plus point with that approach is we don't have to do `into {}`, which might gives a better performance ?